tests/CoreLib/DSet.fs

```diff
@@ -1536,3 +1536,70 @@ type DSetTests () =
         x.SaveToHDDTestWithMonitor 7 9 true
         x.SaveToHDDTestWithMonitor 9 11 true
         x.SaveToHDDTestWithMonitor 11 13 true
+
+    /// At this moment, remote exception will be directly carried back. 
+    /// In the future, we may choose to retry exception at partition (in other node), and only fails the job/partition after repeated retry. 
+    [<Test(Description = "Test: Throw Remote Exception and catch locally")>]
+    member x.DSetThrowRemoteExceptionAndCatchLocally () = 
+        let guid = Guid.NewGuid().ToString("D")
+        let dset = DSet<_> ( Name = guid, Cluster = cluster)
+        let bExceptionCatched = ref false
+        let msg = "Test on Remote Exception"
+        try 
+            let result = (dset |> DSet.source ( fun () -> seq { let ex = System.OperationCanceledException( msg )
+                                                                raise(ex)
+                                                                yield (sprintf "%i-%i" (System.Diagnostics.Process.GetCurrentProcess().Id) 
+                                                                                            (Thread.CurrentThread.ManagedThreadId)) })).ToSeq() |> Array.ofSeq
+
+            Assert.IsNotEmpty(result)
+        with
+        | :? System.AggregateException as ex -> 
+            bExceptionCatched := true
+            for exIn in ex.InnerExceptions do 
+                match exIn with 
+                | :? System.OperationCanceledException as exInner -> 
+                    if String.Compare( exInner.Message, msg, StringComparison.Ordinal) <> 0 then 
+                        bExceptionCatched := false
+                | _ -> 
+                    bExceptionCatched := false
+        | _ -> 
+            ()
+        Assert.AreEqual( !bExceptionCatched, true )
```

Comment:
When this assertion fails, it's unclear it's because the expected exception is not caught, or it's because the msg does not match.